Cargo.toml

```diff
@@ -50,6 +50,7 @@ tokio-rustls = { version = "0.25.0", optional = true }
 tracing = "0.1"
 url = "2"
 semver = { version = "1.0.23", features = ["serde"] }
+futures = "0.3.30"
```

Comment:
The `futures` crate is, to my knowledge, mostly unmaintained these days, so I'd like to avoid taking a dependency on it.

Reply:
And we are also not relying on any of its functionality any longer. Removed